packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -369,6 +397,18 @@ export default class Codec {
     return this._decodeStringMessage(buffer);
   }
 
+  // decodeMfaChallenge decodes a raw tdp MFA challenge message and returns it as a string (of a json).
+  // | message type (10) | mfa_type byte | message_length uint32 | json []byte
+  decodeMfaJson(buffer: ArrayBuffer): MfaJson {
+    let dv = new DataView(buffer);
+    const mfaType = String.fromCharCode(dv.getUint8(1));
+    if (mfaType !== 'n' && mfaType !== 'u') {
+      throw new Error(`invalid mfa type ${mfaType}, should be "n" or "u"`);
```

Comment:
I'm not a fan of throwing as it'll break the app and as mentioned previously there isn't anything 'catching' this. Maybe it should instead return an error that's null otherwise.

Reply:
I don't understand why returning an error would make more sense. We want the app to "break" if this section breaks, in the sense that we want to alert the user that they won't be able to access the desktop session they requested.